main/lsp/LSPPreprocessor.cc

```diff
@@ -125,64 +89,14 @@ void LSPPreprocessor::mergeFileChanges(absl::Mutex &mtx, QueueState &state) {
         }
     }
     ENFORCE(pendingRequests.size() + requestsMergedCounter == originalSize);
-
-    // Check if we should cancel the slow path.
```

Comment:
Goodbye, hairy logic! The preprocessor no longer needs to think about time travel or cancelation.